FormalConjectures/Wikipedia/HadwigerNelson.lean

```diff
@@ -0,0 +1,59 @@
+/-
+Copyright 2025 The Formal Conjectures Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-/
+
+import FormalConjectures.Util.ProblemImports
+
+/-!
+# The Hadwiger–Nelson problem
+
+*Reference:* [Wikipedia](https://en.wikipedia.org/wiki/Hadwiger%E2%80%93Nelson_problem)
+
+*At least 5 colors are required:* [de Grey 2018](https://arxiv.org/abs/1804.02385)
+-/
+
+/--
+The unit-distance graph in the plane, i.e. the graph whose vertices are points in the plane
+and whose edges connect points that are exactly 1 unit apart.
+-/
+def UnitDistancePlaneGraph : SimpleGraph (EuclideanSpace ℝ (Fin 2)) :=
+  SimpleGraph.mk
+    (fun x y => dist x y = 1)
+    (by
+      intros x y
+      simp [dist_comm])
+    (by
+      intros x
+      simp [dist_self])
+/--
+The Hadwiger–Nelson problem asks: How many colors are required to color the plane
+such that no two points at distance 1 from each other have the same color?
+
+It is known that 7 colors are sufficient.
```

Comment:
Would it make sense to also add `UnitDistancePlaneGraph.chromaticNumber ≤ 7` as a sepearate solved theorem?

Reply:
I thought about this, but I was unsure if that's what we were going for. The "proof" is really just this picture from wikipedia, I am not sure if there is a formal source or whether it qualifies as a "research" problem.

![](https://upload.wikimedia.org/wikipedia/commons/thumb/a/a3/Hadwiger-Nelson.svg/480px-Hadwiger-Nelson.svg.png)